android: Update auth URLs to follow convention; Fix JNI callback method signatures

jamilbk opened the pull request:
* Update callbacks and URLs according to #1868 
* Use `SecureRandom`, not `Random` to generate cryptographic tokens
* Minor app manifest cleanup
* Redirect with `client_platform=android` instead of `dest`
* Fix JNI method signatures so that callbacks work

@pratikvelani The AppLinkHandler doesn't seem to be called, because the AuthActivity is still the one that's launched when the `Sign In` button is clicked. Does it make sense to combine those two features into one, or is it better to keep them separate? I'm not an Android developer (obviously), but it seems like there's a lot of boilerplate being duplicated across the two features just to receive an AppLink intent.

jamilbk committed:
synchronize client app auth scheme

jamilbk committed:
Remove client_platform from PortalMock

jamilbk committed:
Fix incomplete classpath in BootShutdownBehavior

jamilbk committed:
BootShutdownReceiver isn't internal

pratikvelani commented on kotlin/android/app/src/main/AndroidManifest.xml:
```diff
@@ -43,19 +43,9 @@
         <activity
             android:name="dev.firezone.android.features.applink.presentation.AppLinkHandlerActivity"
-            android:exported="true"
+            android:exported="false"
```
```suggestion
            android:exported="true"
```
The `exported` attribute exposes, the activity to be launched from outside the app's scope, for ex: launch the app or deeplink. As the AppLinkHandlerActivity handles the applink/deeplink, exported `true` is required.

jamilbk replied:
Thanks for the catch. Apologies for any misunderstanding, but does this mean that any app could open the Firezone AppLink and set the JWT to something invalid in the Firezone app?

I was under the impression that the opened web view was in our app's scope, since it was launched within our app (this is how it works on Apple for example)

jamilbk replied:
Resolved over Slack

jamilbk committed:
Empty BroadcastReceiver constructor

jamilbk committed:
Add @jamilbk's fingerprint to assetlinks.json

jamilbk committed:
Remove ACTION_VIEW intent from AppLink

jamilbk committed:
Revert manifest

jamilbk committed:
Add debug logs around session start

jamilbk committed:
Use explicit return codes in JNI method signatures

This fixes a failure that was occurring silently because the JNI method
signatures did not have their return codes included. The `call_method`
function calls fail, and no error is returned when this happens. For
reference on how to specify JNI method signatures:

https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/types.html

The FFI callbacks are working now, but we're seeing another crash
related to concurrency between Rust and the JVM which will be dealt
with in another PR.

jamilbk committed:
Revert debugging addition
